tsweb/tsweb.go

```diff
@@ -114,3 +111,64 @@ func stripPort(hostport string) string {
 	}
 	return net.JoinHostPort(host, "443")
 }
+
+// varzHandler is an HTTP handler to write expvar values into the
+// prometheus export format:
+//
+//   https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md
+//
+// It makes the following assumptions:
+//
+//   * *expvar.Int are counters.
+//   * an *expvar.Map is descended into, joining keys with
```

Comment:
Why? Prometheus format supports timeseries labels, which is a natural representation for a map.

In particular, timeseries w/ labels support a bunch of natural queries like "sum all labels" or "show me a graph broken down by this label". Denormalizing into discrete timeseries makes querying them as a group harder.

Reply:
Addressed in latest version.